app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(
+            self,
+            name: str,
+            age: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    persons = []
+    for person in people:
+        persons.append(Person(person["name"], person["age"]))
+
+    for person in people:
+        if "wife" in person and person["wife"]:
+            wife = person["wife"]
+            Person.people[person["name"]].wife = Person.people[wife]
+        if "husband" in person and person["husband"]:
+            husband = person["husband"]
+            Person.people[person["name"]].husband = Person.people[husband]
```

Comment:
You are using nested 'if' statements here which is not recommended according to the checklist. Try to avoid nested 'if' by using `and`, `or` logical operators. This will make your code cleaner and easier to understand.